pkg/cmd/run.go

```diff
@@ -68,45 +67,19 @@ getenvoy run postgres:nightly --templateArg endpoints=127.0.0.1:5432,192.168.0.1
 			// If they were passed, config must be created based on
 			// template.
 			if manifestErr == nil && len(templateArgs) > 0 {
-				cmdArg, err := processTemplateArgs(key.Flavor, templateArgs, runtime.(*envoy.Runtime))
+				configYaml, err := flavors.CreateConfig(key.Flavor, templateArgs)
 				if err != nil {
 					return err
 				}
-				args = append(args, cmdArg)
+				args = append(args, `--config-yaml`, configYaml)
 			}
 
 			return runtime.FetchAndRun(args[0], args[1:])
 		},
 	}
 	cmd.Flags().StringToStringVar(&templateArgs, "templateArg", map[string]string{},
 		"arguments passed to a config template for substitution")
+	cmd.PersistentFlags().StringVar(&global.DebugDir, "debug-dir", "",
+		`Store debug data in an alternate directory. Defaults to envoy_home/debug/timestamp`)
 	return cmd
 }
-
-// Function creates config file based on template args passed by a user.
-// The return value is Envoy command line option which must be passed to Envoy.
-func processTemplateArgs(flavor string, templateArgs map[string]string, runtime *envoy.Runtime) (string, error) {
```

Comment:
all this was unnecessary as you can pass a yaml literal. deleting this increases our test coverage.